app/bundles/ReportBundle/Model/ExcelExporter.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\ReportBundle\Model;
+
+use Mautic\CoreBundle\Templating\Helper\FormatterHelper;
+
+/**
+ * Class CsvExporter.
+ */
+class ExcelExporter
+{
+    /**
+     * @var FormatterHelper
+     */
+    protected $formatterHelper;
+
+    public function __construct(FormatterHelper $formatterHelper)
+    {
+        $this->formatterHelper = $formatterHelper;
+    }
+
+    /**
+     * @param array  $reportData
+     * @param string $name
+     *
+     * @throws \Exception
+     */
+    public function export(array $reportData, $name)
+    {
+        if (!class_exists('PHPExcel')) {
+            throw new \Exception('PHPExcel is required to export to Excel spreadsheets');
+        }
+
+        if (!array_key_exists('data', $reportData) || !array_key_exists('columns', $reportData)) {
+            throw new \InvalidArgumentException("Keys 'data' and 'columns' have to be provided");
+        }
+
+        try {
+            $objPHPExcel = new \PHPExcel();
+            $objPHPExcel->getProperties()->setTitle($name);
+
+            $objPHPExcel->createSheet();
+
+            $header = [];
+
+            //build the data rows
+            foreach ($reportData['data'] as $count => $data) {
+                $row = [];
+                foreach ($data as $k => $v) {
+                    if ($count === 0) {
```

Comment:
Is there a way to still set the header columns even when there is no data?

Reply:
It is same problem like with CSV above. We do not have columns if no data provided.